src/odemis/driver/xt_client.py

```diff
@@ -0,0 +1,344 @@
+
+    def set_channel_state(self, name='electron1', state='run'):
+        """
+        Stop or start running the channel. This is non-blocking.
+
+        Parameters
+        ----------
+        name: str
+            name of channel.
+        state: "run" or "stop"
+            desired state of the channel.
+        """
+        self.client.set_channel_state(unicode(name), unicode(state))
+
+    def wait_for_state_changed(self, desired_state, name='electron1', timeout=10):
+        """Wait for channel state to be changed, if it's not changed after 10 seconds break."""
+        self.client.wait_for_state_changed(unicode(desired_state), unicode(name), timeout)
+
+    def get_channel_state(self, name='electron1'):
+        """Return the state of the channel: run, stop or cancel."""
+        return self.client.get_channel_state(unicode(name))
+
+    def get_free_working_distance(self):
+        """Get the free working distance in [m]."""
+        return self.client.get_free_working_distance()
+
+    def set_free_working_distance(self, free_working_distance):
+        """Set the free working distance in [m]."""
+        self.client.set_free_working_distance(free_working_distance)
+
+    def get_fwd_follows_z(self):
+        """
+        Returns True if Z follows free working distance functionality is turned on.
+        When Z follows FWD functionality is switched on and Z stage axis moves, FWD is updated to keep image in focus.
+        """
+        return self.client.get_fwd_follows_z()
+
+    def set_fwd_follows_z(self, follow_z):
+        """
+        Parameters
+        ---------
+        follow_z: bool
+            True if Z follows free working distance functionality should be turned on.
+        """
+        self.client.set_fwd_follows_z(follow_z)
+
+    def set_autofocusing(self, channel, state):
+        """
+        Set the state of autofocus, beam must be turned on. This is non-blocking.
+
+        Parameters
+        ----------
+        channel: xtlib channel object
+            One of the electron channels, the channel must be running.
+        state: "start", "cancel" or "stop"
+            State is start, starts the autofocus. States cancel and stop both stop the autofocusing, some microscopes
+            might need stop, while others need cancel.
+        """
+        self.client.set_autofocusing(channel, unicode(state))
+
+    def is_autofocusing(self):
+        """Returns True if autofocus is running and False if autofocus is not running."""
+        return self.client.is_autofocusing()
+
+    def get_beam_shift(self):
+        """Retrieves the current beam shift x and y values in meter [m]."""
+        return self.client.get_beam_shift()
+
+    def set_beam_shift(self, x_shift, y_shift):
+        """Sets the current beam shift values in meter [m]."""
+        self.client.set_beam_shift(x_shift, y_shift)
+
+    def beam_shift_info(self):
+        """Returns the unit and xy-range of the beam shift."""
```

Comment:
type of return missing. There are multiple places where you only mention the return value in the describing doc string but do not list those separately with the type etc. In some places you do this. Maybe useful to add?